<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>View3dControl</class>
 <widget class="QWidget" name="View3dControl">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>434</width>
    <height>602</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="cameraLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="cameraPositionTextLabel">
       <property name="text">
        <string>Camera:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QDoubleSpinBox" name="cameraXSpinBox">
       <property name="minimum">
        <double>-1000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QDoubleSpinBox" name="cameraYSpinBox">
       <property name="minimum">
        <double>-1000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QDoubleSpinBox" name="cameraZSpinBox">
       <property name="minimum">
        <double>-1000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="cameraViewpointTextLabel">
       <property name="text">
        <string>Viewpoint:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QDoubleSpinBox" name="cameraViewXSpinBox">
       <property name="minimum">
        <double>-1000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QDoubleSpinBox" name="cameraViewYSpinBox">
       <property name="minimum">
        <double>-1000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QDoubleSpinBox" name="cameraViewZSpinBox">
       <property name="minimum">
        <double>-1000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line_7">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="sceneLayout">
     <item row="1" column="0">
      <widget class="QLabel" name="sceneTranslationTextLabel">
       <property name="text">
        <string>Translation:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QDoubleSpinBox" name="sceneXSpinBox">
       <property name="minimum">
        <double>-1000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QDoubleSpinBox" name="sceneYSpinBox">
       <property name="minimum">
        <double>-1000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QDoubleSpinBox" name="sceneZSpinBox">
       <property name="minimum">
        <double>-1000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="sceneRotationTextLabel">
       <property name="text">
        <string>Rotation:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="sceneScaleTextLabel">
       <property name="text">
        <string>Scale:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QDoubleSpinBox" name="sceneYawSpinBox">
       <property name="minimum">
        <double>-180.000000000000000</double>
       </property>
       <property name="maximum">
        <double>179.990000000000009</double>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QDoubleSpinBox" name="scenePitchSpinBox">
       <property name="minimum">
        <double>-180.000000000000000</double>
       </property>
       <property name="maximum">
        <double>179.990000000000009</double>
       </property>
      </widget>
     </item>
     <item row="2" column="3">
      <widget class="QDoubleSpinBox" name="sceneRollSpinBox">
       <property name="minimum">
        <double>-180.000000000000000</double>
       </property>
       <property name="maximum">
        <double>179.990000000000009</double>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QDoubleSpinBox" name="sceneScaleSpinBox">
       <property name="decimals">
        <number>4</number>
       </property>
       <property name="minimum">
        <double>0.000100000000000</double>
       </property>
       <property name="maximum">
        <double>1000000.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="groundLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="groundTextLabel">
       <property name="text">
        <string>Ground:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QDoubleSpinBox" name="groundXSpinBox">
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
       <property name="value">
        <double>30.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QDoubleSpinBox" name="groundYSpinBox">
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
       <property name="value">
        <double>30.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QDoubleSpinBox" name="groundZSpinBox">
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="ColorChooser" name="colorChooser" native="true"/>
   </item>
   <item>
    <widget class="Line" name="line_4">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="showLayout">
     <item row="2" column="0">
      <widget class="QCheckBox" name="showAxesCheckBox">
       <property name="text">
        <string>Show coordinate axes</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QCheckBox" name="showFogCheckBox">
       <property name="text">
        <string>Show background fog</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QCheckBox" name="showGroundCheckBox">
       <property name="text">
        <string>Show ground plane</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="fontLayout">
     <item>
      <widget class="QLabel" name="fontTextLabel">
       <property name="text">
        <string>Font file:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="fontEdit">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="fontBrowseButton">
       <property name="minimumSize">
        <size>
         <width>90</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>Open...</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line_3">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="dumpLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="dumpDirTextLabel">
       <property name="text">
        <string>Dump directory:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="dumpDirEdit">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="dumpDirBrowseButton">
       <property name="minimumSize">
        <size>
         <width>90</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>Browse...</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="dumpFormatTextLabel">
       <property name="text">
        <string>Dump format:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="dumpFormatEdit">
       <property name="inputMask">
        <string/>
       </property>
       <property name="text">
        <string>dump3d%06d.png</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QPushButton" name="dumpButton">
       <property name="minimumSize">
        <size>
         <width>90</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>Dump</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="frameSizeTextLabel">
       <property name="text">
        <string>Dump frame size:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="frameSizeLayout">
       <item>
        <widget class="QSpinBox" name="frameWidthSpinBox">
         <property name="maximum">
          <number>9999</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="frameHeightSpinBox">
         <property name="maximum">
          <number>9999</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line_6">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="dumpAllLayout">
     <item row="0" column="0">
      <widget class="QCheckBox" name="dumpAllCheckBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Dump all frames</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="frameTextLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Frame:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QSpinBox" name="frameSpinBox">
       <property name="minimumSize">
        <size>
         <width>90</width>
         <height>0</height>
        </size>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="maximum">
        <number>999999</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QCheckBox" name="rotateFramesCheckBox">
       <property name="text">
        <string>Rotate between frames</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QDoubleSpinBox" name="frameRotationSpinBox">
       <property name="minimum">
        <double>-180.000000000000000</double>
       </property>
       <property name="maximum">
        <double>179.990000000000009</double>
       </property>
       <property name="singleStep">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ColorChooser</class>
   <extends>QWidget</extends>
   <header>ColorChooser.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>showAxesCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>View3dControl</receiver>
   <slot>showAxesToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dumpDirBrowseButton</sender>
   <signal>clicked()</signal>
   <receiver>View3dControl</receiver>
   <slot>dumpDirBrowseClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>350</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dumpButton</sender>
   <signal>clicked()</signal>
   <receiver>View3dControl</receiver>
   <slot>dumpClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>350</x>
     <y>252</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dumpAllCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>View3dControl</receiver>
   <slot>dumpAllToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>283</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sceneXSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>sceneTranslationChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sceneYSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>sceneTranslationChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>249</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sceneZSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>sceneTranslationChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>347</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sceneYawSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>sceneRotationChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scenePitchSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>sceneRotationChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>249</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sceneRollSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>sceneRotationChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>347</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sceneScaleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>sceneScaleChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showFogCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>View3dControl</receiver>
   <slot>showFogToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>groundXSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>groundRadiusChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showGroundCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>View3dControl</receiver>
   <slot>showGroundToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fontBrowseButton</sender>
   <signal>clicked()</signal>
   <receiver>View3dControl</receiver>
   <slot>fontBrowseClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>356</x>
     <y>29</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>groundYSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>groundRadiusChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>249</x>
     <y>104</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>groundZSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>groundElevationChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>347</x>
     <y>104</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraXSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>cameraPositionChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>196</x>
     <y>17</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraYSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>cameraPositionChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>17</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraZSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>cameraPositionChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>454</x>
     <y>17</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraViewXSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>cameraViewpointChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>196</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraViewYSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>cameraViewpointChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraViewZSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>View3dControl</receiver>
   <slot>cameraViewpointChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>454</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>dumpDirBrowseClicked()</slot>
  <slot>dumpClicked()</slot>
  <slot>dumpAllToggled(bool)</slot>
  <slot>showAxesToggled(bool)</slot>
  <slot>sceneTranslationChanged()</slot>
  <slot>sceneRotationChanged()</slot>
  <slot>sceneScaleChanged()</slot>
  <slot>showFogToggled(bool)</slot>
  <slot>groundRadiusChanged(double)</slot>
  <slot>showGroundToggled(bool)</slot>
  <slot>fontBrowseClicked()</slot>
  <slot>groundElevationChanged(double)</slot>
  <slot>cameraPositionChanged()</slot>
  <slot>cameraViewpointChanged()</slot>
 </slots>
</ui>
